### PR TITLE
Solution for 021_MergeTwoSortedLists

### PR DESCRIPTION
**Pull request by Jovannna**
Again I made some mistakes when commiting. I commited on the wrong branch :( But i think this pull request is ok.

### CODE REVIEW

File: 021_MergeTwoSortedLists [hunk 1]
```diff
@@ -0,0 +1,80 @@
+ class ListNode(object):
```
Comment (andrejkoilic) [inline 021_MergeTwoSortedLists]: I assume this file is added by mistake, you have both "021_MergeTwoSortedLists" and "021_MergeTwoSortedLists.py"

File: 021_MergeTwoSortedLists.py [hunk 2]
```diff
@@ -0,0 +1,84 @@
+class ListNode(object):
+     def __init__(self, x):
+         self.val = x
+         self.next = None
+class ListNode(object):
```
Comment (andrejkoilic) [inline 021_MergeTwoSortedLists.py]: You have class ListNode twice

File: 021_MergeTwoSortedLists.py [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    for i in range(len(arr)):
+        if l==None:
+            l=ListNode(arr[i])
+            last=l
+        else:
+            pom=ListNode(arr[i])
+            last.next=pom
+            last=last.next
+    return l
+def listToArray(l):
```
Comment (andrejkoilic) [inline 021_MergeTwoSortedLists.py]: It makes more sense for this to be function in ListNode, as you are actually here working on instance "l" of ListNode

File: 021_MergeTwoSortedLists.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+         self.next = None
+class ListNode(object):
+     def __init__(self, x):
+         self.val = x
+         self.next = None
+
+class Solution(object):
+    def mergeTwoLists(self, l1, l2):
+        last=None
+        lr=None
```
Comment (andrejkoilic) [inline 021_MergeTwoSortedLists.py]: Change lr to more descriptive variable name.

File: 021_MergeTwoSortedLists.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+        while (l1!=None) and (l2!=None):
+            if l1.val<=l2.val:
+                if lr==None:
+                    lr=l1
+                    last=l1
+                else: 
+                    last.next=l1
+                    last=last.next
```
Comment (andrejkoilic) [inline 021_MergeTwoSortedLists.py]: line 18 and line 21 are doing the same thing, "last = l1". You can remove that from if/else and add it outside before the "l1 = l1.next"

Comment (andrejkoilic) [reply 021_MergeTwoSortedLists.py]: Same for the 2nd part, when you are looking into l2

File: 021_MergeTwoSortedLists.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+                if (l2.val<l1.val):
+                    if lr==None:
+                        lr=l2
+                        last=lr
+                    else: 
+                        last.next=l2 
+                        last=last.next
+                l2=l2.next
+                       
+        if (l1==None):
```
Comment (andrejkoilic) [inline 021_MergeTwoSortedLists.py:36]: Try adding some comments. For example here you are checking if there is some tail in L1 or L2 and just reconnecting that tail to result list.

File: 021_MergeTwoSortedLists.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+            last.next=l2
+        else:
+            if (l2==None):
```
Comment (andrejkoilic) [inline 021_MergeTwoSortedLists.py]: Do you need this check?

File: 021_MergeTwoSortedLists.py [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+                last.next=l1
+        return lr
+            
+def arrayToList(arr):
+    l=None
```
Comment (andrejkoilic) [inline 021_MergeTwoSortedLists.py:43]: Use more descriptive variable names

**Review by andrejkoilic — CHANGES_REQUESTED** [review]

File: 021_MergeTwoSortedLists.py (hunk 3 above)
Comment (Jovannna) [reply 021_MergeTwoSortedLists.py]: I'm not sure I did this right way.

Comment (Jovannna) [reply 021_MergeTwoSortedLists.py]: I corrected it.

File: 021_MergeTwoSortedLists.py [hunk 9]
```diff
@@ -0,0 +1,80 @@
+class ListNode(object):
+    def __init__(self, x):
+        self.val = x
+        self.next = None
+    def listToArray(self):
```
Comment (andrejkoilic) [inline 021_MergeTwoSortedLists.py:5]: nit: When you have a class method, it's good practice to document your function. For this, use """ <explanation> """ docstring. For example, for more info take a look at: https://realpython.com/documenting-python-code/

File: 021_MergeTwoSortedLists.py [hunk 10]
```diff
@@ -0,0 +1,80 @@
...
+            last=last.next
+    return l
+
+
+
+def test(inputList1, inputList2, outputList):
+    
+    sol = Solution()
+    list1=arrayToList(inputList1)
+
```
Comment (andrejkoilic) [inline 021_MergeTwoSortedLists.py:58]: Remove extra lines, 1 line as separator is sufficient.

File: 021_MergeTwoSortedLists (hunk 1 above)
Comment (andrejkoilic) [reply 021_MergeTwoSortedLists]: You still have this file, you should delete it from the change.

File: 021_MergeTwoSortedLists.py [hunk 11]
```diff
@@ -0,0 +1,80 @@
...
+    
+    sol = Solution()
+    list1=arrayToList(inputList1)
+    
+
+    list2=arrayToList(inputList2)
+    
+    list3=sol.mergeTwoLists(list1,list2)
+
+    inputList=list3.listToArray()
```
Comment (andrejkoilic) [inline 021_MergeTwoSortedLists.py]: inputList variable name is confusing, this is more as resultList or something.

**Review by andrejkoilic — APPROVED** [review]

**Pull request merged**